src/pages/build-apps/guides/transaction-signing.md

```diff
@@ -0,0 +1,260 @@
+| appDetails      | object   | true     | Dictionary that requires `name` and `icon` for app                                                                                                                                                   |
+| finished        | function | true     | Callback executed by app when transaction has been signed and broadcasted. It received an object back with `txId` and `txRaw` properties, both of which are strings.                                 |
+| authOrigin      | string   | false    | URL of authenticator to use for prompting signature and broadcast. Defaults to `https://wallet.hiro.so` for the Stacks Wallet, which is handled by the Stacks Wallet browser extension if installed. |
+
+### Passing Clarity types with function arguments
+
+To serialize your transaction properly, you'll need to provide an appropriate Clarity type for each function argument.
+
+These types are named the same as in Clarity and must be passed as strings:
+
+- `uint`: Unsigned integer (e.g. `"240"`)
+- `int`: Signed integer (e.g. `"12"`)
+- `bool`: Boolean (e.g. `true` or `false`, `0` or `1`)
+- `buff`: Buffer (e.g. `"hello world"`)
+- `principal`: [Contract or standard principal](/write-smart-contracts/principals) (e.g. `"ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ"`) or `"ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ.my-contract"`.
+
+Argument are provided in an array of objects with keys `type` and `value`:
+
+```tsx
+const functionArguments = [
+  {
+    type: 'buff',
+    value: 'hello, world',
+  },
+  {
+    type: 'uint',
+    value: '1',
+  },
+];
+```
+
+-> If you're using TypeScript, these Clarity types can be imported as `ContractCallArgumentType` from `@stacks/connect`.
+
+## Usage in React Apps
+
+Import the `useConnect` from the `connect` package to integrate transaction signing more seamlessly into React apps.
+
+Each transaction signing method is itself available as a function returned by `useConnect` though prefixed with `do` for consistency with React action naming standards:
+
+- `openContractCall` as `doContractCall`
+- `openSTXTransfer` as `doSTXTransfer`
+- `openContractDeploy` as `doContractDeploy`
+
+Use these functions with the same parameters as outlined above. However, you don't have to specify `appDetails` since they are detected automatically if `useConnect` has been used already for authentication.
+
+```tsx
+import { useConnect } from '@stacks/connect';
+
+const MyComponent = () => {
+  const { doContractCall } = useConnect();
+
+  const onClick = async () => {
+    const options = {
+      /** See examples above */
+    };
+    await doContractCall(options);
+  };
+
+  return <span onClick={onClick}>Call my contract</span>;
+};
+```
+
+## Change network
+
+TBD: Instructions for changing between mainnet, testnet and local networks
```

Comment:
@hstove Mind providing a code snippet here for changing the network used by the app and authenticator for transaction signing?

Reply:
The deleted docs I had: https://github.com/blockstack/docs/commit/a6cbe412713dd414c0ea7e18cf95925d84b0429f#diff-cb5085f3a080ffe5726cf03a9a1ba1eafac4cda786e5b482051dfdc47c04267bL110